faunadb-scala/src/main/scala/faunadb/values/Time.scala

```diff
@@ -0,0 +1,102 @@
+package faunadb.values
+package time
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+/**
+  * Represents the instant from Java's epoch composed by the number of seconds since eposh
+  * and the nanoseconds offset.
+  */
+class HighPrecisionTime private(val secondsSinceEpoch: Long, val nanoSecondsOffset: Int) {
+
+  /** Returns a [[org.joda.time.Instant]]. Truncates micro and nanoseconds. */
+  def toInstant: Instant =
+    new Instant(milliSecondsSinceEpoch)
+
+  /** Milliseconds since Java epoch. Truncates micro and nanoseconds. */
+  def milliSecondsSinceEpoch: Long =
+    secondsSinceEpoch * 1000
+
+  override def equals(obj: Any): Boolean = obj match {
+    case other: HighPrecisionTime =>
+      secondsSinceEpoch == other.secondsSinceEpoch &&
+        nanoSecondsOffset == other.nanoSecondsOffset
+
+    case _ => false
+  }
+
+  override def hashCode(): Int =
+    secondsSinceEpoch.hashCode() + nanoSecondsOffset.hashCode()
```

Comment:
also, scala style nit: hashCode is pure, has no side effects. it should be called w/o parens.